src/b3d/chisight/gen3d/image_kernel.py

```diff
@@ -310,3 +310,162 @@ def get_rgbd_vertex_kernel(self) -> PixelRGBDDistribution:
         # they don't expect observed_rgbd to be invalid, so we need to handle
         # that manually.
         return self.rgbd_vertex_kernel
+
+
+### Unique point per pixel image kernel ###
+
+
+def calculate_latent_and_observed_correspondences(
```

Comment:
I factored this out so we can use it to establish the same pixel<>point correspondences in inference as in the model.